### PR TITLE
Refactor keyVault service... because?

### PR DESCRIPTION
**Pull request by glenjamin**

### CODE REVIEW

File: server/services/keyVault.js [hunk 1]
```diff
@@ -26,95 +25,77 @@ function getKeyVaultCredentials() {
   return Promise.resolve(createVaultCredentials());
 }
 
-async function createKeyVaultService() {
-  const credentials = await getKeyVaultCredentials();
-  const client = new KeyVaultClient(credentials);
+async function createKeyVaultService(override) {
+  const client = override || new KeyVaultClient(await getKeyVaultCredentials());
```
Comment (romidane) [inline server/services/keyVault.js]: I'd probably split this up to reduce overload

File: server/services/keyVault.js [hunk 2]
```diff
@@ -26,95 +25,77 @@ function getKeyVaultCredentials() {
...
-      return { ok: authenticated, data: (authenticated) ? data : null };
+      return {
+        ok: authenticated,
+        data: (authenticated) ? { expires: attributes.expires } : null,
+      };
     } catch (error) {
-      logger.error(error);
-      return { ok: false, data: null };
+      logger.error(error, 'Failed to validate user');
+      return false;
```
Comment (romidane) [inline server/services/keyVault.js]: can't be returning different types from a function.. leads to an unstable api

Comment (glenjamin) [reply server/services/keyVault.js]: Good spot, i think i missed this one during the merge.

File: server/services/keyVault.js (hunk 1 above)
Comment (glenjamin) [reply server/services/keyVault.js]: as in multiple lines? The main reason I didn't was to avoid having to use `let`

**Pull request merged**